pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -218,33 +221,60 @@ func (m FlavorAssignmentMode) String() string {
 }
 
 type FlavorAssignment struct {
-	Name   kueue.ResourceFlavorReference
-	Mode   FlavorAssignmentMode
-	borrow int64
+	Name                  kueue.ResourceFlavorReference
+	Mode                  FlavorAssignmentMode
+	LastAssignedFlavorIdx int
+	borrow                int64
 }
 
 // AssignFlavors assigns flavors for each of the resources requested in each pod set.
 // The result for each pod set is accompanied with reasons why the flavor can't
 // be assigned immediately. Each assigned flavor is accompanied with a
 // FlavorAssignmentMode.
 func AssignFlavors(log logr.Logger, wl *workload.Info, resourceFlavors map[kueue.ResourceFlavorReference]*kueue.ResourceFlavor, cq *cache.ClusterQueue, counts []int32) Assignment {
+	if wl.LastAssignment != nil {
+		if cq.Generation > wl.LastAssignment.ClusterQueueGeneration &&
+			(cq.Cohort == nil || cq.Cohort.Generation > wl.LastAssignment.CohortGeneration) {
```

Comment:
Shouldn't it be like this?
```suggestion
		if cq.Generation > wl.LastAssignment.ClusterQueueGeneration ||
			(cq.Cohort != nil && cq.Cohort.Generation > wl.LastAssignment.CohortGeneration) {
```

Please make sure to have a unit test for this.

Reply:
Sure. I will add a unit test for this.